### ISSUE TITLE
Creditor gets reset in multi-creditor environments

### ISSUE THREAD

**Issue by patf** [Project60/org.project60.sepa#505]
When CiviSEPA closes transaction groups of type OOF or FRST, it updates the mandate status to SENT or RCUR respectively. In environments with multiple SEPA creditors, because the API does not check whether it is creating a new mandate or updating an existing one, this update causes the creditor to be set to the default creditor, even when a different creditor was selected previously.

**Comment by systopia** [comment]
@patf Thanks for reporting/fixing. I have adjusted the fix slightly (see [dev_505_master](https://github.com/Project60/org.project60.sepa/tree/dev_505_master)), could you have another look please?

**Comment by patf** [comment]
That makes sense to cover cases where someone uses the BAO class directly. LGTM!

**Comment by systopia** [comment]
Merged compiled fix to ``master``, @patf  could you verify plz?

**Comment by patf** [comment]
Fix is verified for master and the 1.2 branch.

**Comment by systopia** [comment]
FYI: Here's a quick SQL query to see whether it has happened in your environment:
```
SELECT 
 mandate.reference           AS reference,
 SUBSTR(txg.reference, 5, 1) AS group_creditor,
 mandate.creditor_id         AS mandate_creditor
FROM civicrm_sdd_mandate mandate
LEFT JOIN civicrm_contribution contribution     ON contribution.contribution_recur_id = mandate.entity_id
LEFT JOIN civicrm_sdd_contribution_txgroup ctxg ON ctxg.contribution_id = contribution.id
LEFT JOIN civicrm_sdd_txgroup              txg  ON txg.id = ctxg.txgroup_id
WHERE mandate.entity_table = 'civicrm_contribution_recur'
  AND SUBSTR(txg.reference, 5, 1) <> mandate.creditor_id
GROUP BY mandate.id;
```

Be aware that this only works on RCUR mandates, and is not necessarily complete.
